debezium-core/src/test/java/io/debezium/config/ConfigurationTest.java

```diff
@@ -339,6 +341,22 @@ public void testServerNameValidation() {
                         Field.validationOutput(TOPIC_PREFIX, "server@X has invalid format (only the underscore, hyphen, dot and alphanumeric characters are allowed)"));
     }
 
+    @Test
+    @FixFor("DBZ-6156")
+    public void testHostnameValidation() {
+        config = Configuration.create().with(HOSTNAME, "").build();
+        assertThat(config.validate(Field.setOf(HOSTNAME)).get(HOSTNAME.name()).errorMessages().get(0))
+                .isEqualTo("The 'database.hostname' value is invalid: A value is required");
+
+        List<String> invalidHostnames = Collect.arrayListOf("~hostname", "hostname@", "host-*-name", "(hostname)");
```

Comment:
Should we test for `hostname?inject_parameter=1234` too? @jpechane

Reply:
@rk3rn3r Thanks for the recomendation! https://github.com/debezium/debezium/pull/4331